mod.d/enadiag.yaml

```
@@ -0,0 +1,270 @@
+                                                universal_newlines=True, stderr=subprocess.STDOUT)
+      except Exception as ex:
+          print(ex)
+          print("[WARN] Unable to get stats from ethool")
+      return stat_output
+
+  def save_ena_stats():
+      """
+      Regexes out the statistic values we care about and adds them to the Interface class object.
+      """
+      # Simple regex on each stat name and values, pull value group
+      try:
+          for interface in interface_results:
+              stat_output = get_ena_stats(interface.name)
+              interface.tx_timeout = re.search("(tx_timeout: )(\d+)", stat_output).group(2)
+              interface.suspend = re.search("(suspend: )(\d+)", stat_output).group(2)
+              interface.wd_expired = re.search("(wd_expired: )(\d+)", stat_output).group(2)
+              interface.interface_down = re.search("(interface_down: )(\d+)", stat_output).group(2)
+              interface.admin_q_pause = re.search("(admin_q_pause: )(\d+)", stat_output).group(2)
+              interface.bw_in_allowance_exceeded = re.search("(bw_in_allowance_exceeded: )(\d+)", stat_output).group(2)
+              interface.bw_out_allowance_exceeded = re.search("(bw_out_allowance_exceeded: )(\d+)", stat_output).group(2)
+              interface.pps_allowance_exceeded = re.search("(pps_allowance_exceeded: )(\d+)", stat_output).group(2)
+              interface.conntrack_allowance_exceeded = re.search("(conntrack_allowance_exceeded: )(\d+)",
+                                                                 stat_output).group(2)
+              interface.linklocal_allowance_exceeded = re.search("(linklocal_allowance_exceeded: )(\d+)",
+                                                                 stat_output).group(2)
+      except Exception as ex:
+          print(ex)
+          print("[WARN] Unable to save interface stats to class object.")
```

- The warn message  is kind of ambiguous.
- Seems this would only happen if the ethtool output changes, right?
- Should the entire function fail if only one field is missing?

---

Could we clarify the warn message then? Something along the lines of "expected ethtool output perhaps?

---

Sure can!

---

Can do.